python/node_classification.py

```diff
@@ -0,0 +1,421 @@
+
+
+@mgp.read_proc
+def set_model_parameters(
+    ctx: mgp.ProcCtx, params: mgp.Map = {}
+) -> mgp.Record(in_channels=int, out_channels=int, num_samples=int, path=str):
+    """The purpose of this function is to initialize all global variables. Parameter
+    params is used for variables written in query module. It first checks
+    if (new) variables in params are defined appropriately. If so, map of default
+    global parameters is overriden with user defined dictionary params.
+    After that it executes previously defined functions declare_globals and
+    declare_model_and_data and sets each global variable to some value.
+
+    Args:
+        ctx: (mgp.ProcCtx): current context,
+        params: (mgp.Map, optional): user defined parameters from query module. Defaults to {}
+
+    Raises:
+        Exception: exception is raised if some variable in dictionary params is not
+                    defined as it should be
+
+    Returns:
+        mgp.Record(
+            in_channels (int): number of input channels,
+            out_channels (int): number of output channels
+            num_sample(int): number of samples
+        )
+    """
+    global DEFINED_INPUT_TYPES, DEFAULT_VALUES
+
+    # function checks if input values in dictionary are correctly typed
+    def is_correctly_typed(defined_types, input_values):
+        if isinstance(defined_types, dict) and isinstance(input_values, dict):
+            # defined_types is a dict of types
+            return all(
+                k in input_values  # check if exists
+                and is_correctly_typed(
+                    defined_types[k], input_values[k]
+                )  # check for correct type
+                for k in defined_types
+            )
+        elif isinstance(defined_types, type):
+            return isinstance(input_values, defined_types)
+        else:
+            return False
+
+    # mgconsole bug
+    if (
+        ModelParams.HIDDEN_FEATURES_SIZE in params.keys()
+        and type(params[ModelParams.HIDDEN_FEATURES_SIZE]) == tuple
+    ):
+        params[ModelParams.HIDDEN_FEATURES_SIZE] = list(
+            params[ModelParams.HIDDEN_FEATURES_SIZE]
+        )
+    if (
+        DataParams.METRICS in params.keys()
+        and type(params[DataParams.METRICS]) == tuple
+    ):
+        params[DataParams.METRICS] = list(params["metrics"])
+
+    params = {**DEFAULT_VALUES, **params}  # override any default parameters
+
+    print(params)
+
+    if not is_correctly_typed(DEFINED_INPUT_TYPES, params):
+        raise Exception(
+            f"Input dictionary is not correctly typed. Expected following types {DEFINED_INPUT_TYPES}."
+        )
+
+    declare_globals(params)
+    declare_model_and_data(ctx)
+
+    return mgp.Record(
+        in_channels=global_params[ModelParams.IN_CHANNELS],
+        out_channels=global_params[ModelParams.OUT_CHANNELS],
+        num_samples=np.shape(Modelling.data.x)[0],
+        path=global_params[OtherParams.PATH_TO_MODEL],
+    )
```

Comment:
This isn't the information the user expects to see once he/she sets the parameters

Reply:
Okay, I'll change them. Thanks!

Reply:
Repaired.